File: leetcode/top150/80-remove-duplicates-sorted-arrays-ii/python/solution.py
import pdb
from templates.python.solution import SolutionBase

class Solution(SolutionBase):

    def __init__(self, *inputs):
        super().__init__(*inputs)

    def solution(self):
        c_val = self.inputs[0][0]
        i = 1
        j = 1

        while (i < len(self.inputs[0])):
            if (c_val == self.inputs[0][i]):
                if (j == 2):
                    self.inputs[0].pop(i)
                    continue
                i = i + 1
                j = j + 1
                continue
            c_val = self.inputs[0][i]
            i = i + 1
            j = 1

        self.logger.info(f'self.inputs[0]: {self.inputs[0]}')
        return len(self.inputs[0])

File: leetcode/top150/80-remove-duplicates-sorted-arrays-ii/c++/solution.cpp
#include "templates/solution.hpp"

using namespace solution;

// Fill in this solution if a return value is expected
// Provide input/output types to function declaration
template<>
void Solution::solution(std::vector<std::shared_ptr<output<int>>>& outputs,
                        std::vector<std::shared_ptr<input<std::vector<int>>>>& inputs) {
    
    int c_val = std::get<0>(inputs[0]->inputs)[0];
    size_t i = 1;
    int j = 1;
    
    while (i < std::get<0>(inputs[0]->inputs).size()) {
        if (c_val == std::get<0>(inputs[0]->inputs)[i]) {
            if (j == 2) {
                auto it = std::get<0>(inputs[0]->inputs).begin() + i;
                std::get<0>(inputs[0]->inputs).erase(it);
                continue;
            }
            ++i;
            ++j;
            continue;
        }
        c_val = std::get<0>(inputs[0]->inputs)[i];
        ++i;
        j = 1;
    }

    std::tuple<int> count = std::make_tuple(std::get<0>(inputs[0]->inputs).size());
    outputs[0]->outputs.swap(count);

    return;
}

// Fill in this solution if no return value is expected
// Provide input types to function declaration
template<>
void Solution::inplace_solution(std::vector<std::shared_ptr<input<std::vector<int>>>>& inputs) {
    return;
}